File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;
class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        HashMap<Character, Integer> map = new HashMap<>();
        for(int i = 0; i < survey.length; i++) {
            if(choices[i] < 4) {
                char key = survey[i].charAt(0);
                int value = map.getOrDefault(key,0);
                int newValue = value + 4 - choices[i];
                map.put(key,newValue);
            }
            else {
                char key = survey[i].charAt(1);
                int value = map.getOrDefault(key,0);
                int newValue = value +  choices[i] - 4;
                map.put(key,newValue);
            }
        }
        for(char key : map.keySet()) {
            System.out.print(key);
        }
        if(map.getOrDefault('R',0) >= map.getOrDefault('T',0)) {
            answer += 'R';
        }
        else {
            answer += 'T';
        }
         if(map.getOrDefault('C',0) >= map.getOrDefault('F',0)) {
            answer += 'C';
        }
        else {
            answer += 'F';
        }
         if(map.getOrDefault('J',0) >= map.getOrDefault('M',0)) {
            answer += 'J';
        }
        else {
            answer += 'M';
        }
         if(map.getOrDefault('A',0) >= map.getOrDefault('N',0)) {
            answer += 'A';
        }
        else {
            answer += 'N';
        }
        return answer;
    }
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    var answer = '';
    const obj = {};
    const arr = ['RT','CF','JM','AN'];
    for (const element of arr) {
        const [s1, s2] = element.split('');
        obj[s1] = 0;
        obj[s2] = 0;
    }
    const MID = 4;
    for (let i = 0; i < survey.length; i++) {
        const [s1,s2] = survey[i].split('');
        const choice = choices[i];
        if (choice < MID) {
            obj[s1] = obj[s1] + (MID - choice);
        } else {
            obj[s2] = obj[s2] + (choice - MID);
        }
    }
    for (const element of arr) {
        const [s1,s2] = element.split('');
        if (obj[s1] > obj[s2]) {
            answer += s1;
        } else if (obj[s1] < obj[s2]) answer += s2;
        else {
            if (s1.localeCompare(s2) < 0) answer += s1;
            else answer += s2;
        }
    }
    return answer;
}